packages/cli/src/controllers/users.controller.ts

```diff
@@ -64,20 +69,16 @@ export class UsersController {
 
 	private removeSupplementaryFields(
 		publicUsers: Array<Partial<PublicUser>>,
-		listQueryOptions: ListQuery.Options,
+		listQueryOptions: UsersListFilterDto,
 	) {
-		const { take, select, filter } = listQueryOptions;
+		const { select } = listQueryOptions;
 
 		// remove fields added to satisfy query
 
-		if (take && select && !select?.id) {
+		if (select !== undefined && !select.includes('id')) {
 			for (const user of publicUsers) delete user.id;
 		}
 
-		if (filter?.isOwner) {
-			for (const user of publicUsers) delete user.role;
-		}
-
 		// remove computed fields (unselectable)
 
 		if (select) {
```

Comment:
We added isOwner, isPending, singinType in the response schema no ? Why removing them ?

Reply:
These are removed if a select parameter was given, in which case they couldn't have been selected, because they are not part of the select schema. Since they are computed fields in the client, we can't select them in the database.